g4root/src/TG4RootDetectorConstruction.cxx

```diff
@@ -374,31 +374,71 @@ G4Material* TG4RootDetectorConstruction::CreateG4Material(
     pMaterial =
       new G4Material(name, density, nComponents, state, temp, pressure);
     for (Int_t i = 0; i < nComponents; i++) {
-      //         TGeoElement *elem = mixt->GetElement(i);
-      //         name = elem->GetTitle();
-      //         G4Element *pElement = G4Element::GetElement(name);
-      TGeoElement* elem = table->GetElement(Int_t(mixt->GetZmixt()[i]));
-      if (!elem) {
-        G4ExceptionDescription description;
-        description << "      "
-                    << "Woops: no element corresponding to Z="
-                    << Int_t(mixt->GetZmixt()[i]);
-        G4Exception("TG4RootDetectorConstruction::CreateG4Material",
-          "G4Root_F006", FatalException, description);
+      TGeoElement *elem = mixt->GetElement(i);
+      // name = elem->GetTitle();
+      // G4Element *pElement = G4Element::GetElement(name);
+      // TGeoElement* elem = table->GetElement(Int_t(mixt->GetZmixt()[i]));
+      // if (!elem) {
+      //  G4ExceptionDescription description;
+      //  description << "      "
+      //              << "Woops: no element corresponding to Z="
+      //              << Int_t(mixt->GetZmixt()[i]);
+      //  G4Exception("TG4RootDetectorConstruction::CreateG4Material",
+      //    "G4Root_F006", FatalException, description);
+      //    }
+      G4Element* pElement;
+      if (Int_t nIsotopes = elem->GetNisotopes(); nIsotopes > 0) { // user-defined element with isotopes
+        for (Int_t i = 0; i < nIsotopes; i++) {
+          TGeoIsotope *rIso = elem->GetIsotope(i);
+          if (i == 0) {
+            elname = rIso->GetTitle();
+            symbol = table->GetElement(rIso->GetZ())->GetName();
+            pElement = new G4Element(elname, symbol, nIsotopes);
+          }
+          G4Isotope *pIso = new G4Isotope(rIso->GetName(), rIso->GetZ(), rIso->GetN(), rIso->GetA() * (g / mole));
+          pElement->AddIsotope(pIso, elem->GetRelativeAbundance(i));
+        }
+      } else {
```

Comment:
As this code is repeated once again, it would be better to add a protected function
```cpp
  G4Element* CreateG4Element(const TGeoElement* elem);
```
and move the code here.